community/kernel/src/main/java/org/neo4j/kernel/api/ExecutingQuery.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api;
+
+import java.util.Map;
+
+public class ExecutingQuery
+{
+    private final long queryId;
+
+    private final String authSubjectName;
+    private final String queryText;
+    private final Map<String, Object> queryParameters;
+    private final long startTime;
+
+    // TODO: Metadata
+
+    public ExecutingQuery( long queryId, String authSubjectName, String queryText, Map<String,Object> queryParameters )
+    {
+        this.queryId = queryId;
+        this.authSubjectName = authSubjectName;
+        this.queryText = queryText;
+        this.queryParameters = queryParameters;
+        this.startTime = System.currentTimeMillis();
```

Comment:
can we please propagate clocks here and take time from them instead of using system time directly